src/modules/hip/kernel/glitch.hpp

```diff
@@ -0,0 +1,299 @@
+{
+    d_float8 increment_f8, rLocsX_f8, rLocsY_f8, gLocsX_f8, gLocsY_f8, bLocsX_f8, bLocsY_f8;
+    increment_f8.f4[0] = make_float4(0.0f, 1.0f, 2.0f, 3.0f);
+
+    rLocsX_f8.f4[0] = static_cast<float4>(id_x + rgbOffsets.r.x) + increment_f8.f4[0];
+    rLocsX_f8.f4[1] = rLocsX_f8.f4[0] + (float4) 4;
+    rLocsY_f8.f4[0] = rLocsY_f8.f4[1] = static_cast<float4>(id_y + rgbOffsets.r.y);
+    check_locs(rLocsX_f8.f4[0], rLocsY_f8.f4[0], rgbOffsets.r, roiTensorPtrSrc);
+    check_locs(rLocsX_f8.f4[1], rLocsY_f8.f4[1], rgbOffsets.r, roiTensorPtrSrc);
+
+    gLocsX_f8.f4[0] = static_cast<float4>(id_x + rgbOffsets.g.x) + increment_f8.f4[0];
+    gLocsX_f8.f4[1] = gLocsX_f8.f4[0] +(float4) 4;
+    gLocsY_f8.f4[0] = gLocsY_f8.f4[1]  = static_cast<float4>(id_y + rgbOffsets.g.y);
+    // gLocsY_f8.f4[1] = static_cast<float4>(id_y + rgbOffsets.g.y);
+    check_locs(gLocsX_f8.f4[0], gLocsY_f8.f4[0], rgbOffsets.g, roiTensorPtrSrc);
+    check_locs(gLocsX_f8.f4[1], gLocsY_f8.f4[1], rgbOffsets.g, roiTensorPtrSrc);
+
+    bLocsX_f8.f4[0] = static_cast<float4>(id_x + rgbOffsets.b.x) + increment_f8.f4[0];
+    bLocsX_f8.f4[1] = bLocsX_f8.f4[0] + (float4) 4;
+    bLocsY_f8.f4[0] = bLocsY_f8.f4[1] = static_cast<float4>(id_y + rgbOffsets.b.y);
+    //  = static_cast<float4>(id_y + rgbOffsets.b.y);
+    check_locs(bLocsX_f8.f4[0], bLocsY_f8.f4[0], rgbOffsets.b, roiTensorPtrSrc);
+    check_locs(bLocsX_f8.f4[1], bLocsY_f8.f4[1], rgbOffsets.b, roiTensorPtrSrc);
+
+    rlocSrc_f16->f4[0] = rLocsX_f8.f4[0];
+    rlocSrc_f16->f4[1] = rLocsX_f8.f4[1];
+    rlocSrc_f16->f4[2] = rLocsY_f8.f4[0];
+    rlocSrc_f16->f4[3] = rLocsY_f8.f4[1];
+
+    glocSrc_f16->f4[0] = gLocsX_f8.f4[0];
+    glocSrc_f16->f4[1] = gLocsX_f8.f4[1];
+    glocSrc_f16->f4[2] = gLocsY_f8.f4[0];
+    glocSrc_f16->f4[3] = gLocsY_f8.f4[1];
+
+    blocSrc_f16->f4[0] = bLocsX_f8.f4[0];
+    blocSrc_f16->f4[1] = bLocsX_f8.f4[1];
+    blocSrc_f16->f4[2] = bLocsY_f8.f4[0];
+    blocSrc_f16->f4[3] = bLocsY_f8.f4[1];
+
+}
+
+template <typename T>
+__global__ void glitch_pkd_hip_tensor(T *srcPtr,
+                                      uint2 srcStridesNH,
+                                      T *dstPtr,
+                                      uint2 dstStridesNH,
+                                      RpptChannelOffsets *rgbOffsetsPtr,
+                                      RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+    {
+        return;
+    }
+
+    uint srcIdx = (id_z * srcStridesNH.x);
+    uint dstIdx = (id_z * dstStridesNH.x) + (id_y * dstStridesNH.y) + id_x * 3;
+
+    int4 srcRoi_i4 = *(reinterpret_cast<int4 *>(&roiTensorPtrSrc[id_z]));
```

Comment:
Why do we need a reinterpret case here?
cannot we use them as uint in the code?

Reply:
we are using int4 in all other HIP kernels also